api/v1alpha1/nicclusterpolicy_types.go

```diff
@@ -67,8 +67,15 @@ type ConfigMapNameReference struct {
 	Name string `json:"name,omitempty"`
 }
 
+// InitContainerSpec contains configuration for the init container of the OFED driver
+type InitContainerSpec struct {
+	ImageSpec `json:""`
+}
+
 // OFEDDriverSpec describes configuration options for OFED driver
 type OFEDDriverSpec struct {
+	// Contains spec for the init container
+	InitContainer *InitContainerSpec `json:"initContainer,omitempty"`
```

Comment:
do we really want customer to be able to configure it ?
maybe keep it "internal" i.e an env var in operator ?

Reply:
one open question is image pull secrets for this container in this case.

Reply:
I don't like that In this case we will have a hidden image which is not a part of the API as all other images. This can confuse users, and for example cause issues with air gap deployments.

Reply:
i think air-gap should not be the lead factor for how we define the api.

you can easily expose the image name as helm chart parameter.
like in sriov network operator.

mofed spec starts to get annoyingly complex which is why id like to simplify.
moreover any change to this parameter will redeploy mofed in the whole cluster better not to expose this knob to the user "easily"

Reply:
ok, let's remove this config from CR. I updated the PR